### PR TITLE
Update CI for 5.x branch to run with multiple Hazelcast versions

### LINKED ISSUE #199: Vert.x 5 : test with Hazelcast 5.5
Given Vert.x 5 requires Java 11 as a minimum version, we cannot make HZ 5.4 or 5.5 the default version (these require Java 17).

However, we may be able to configure the CI to run tests with HZ 5.5 and JDK 17.

The `5.5.0-DEVEL-8` is the latest version available publicly in https://repository.hazelcast.com/devel

The 5.5.0 is expected to be released on July 29th.

### PR DESCRIPTION
**Pull request by frant-hartm**
Motivation:

Test with multiple Hazelcast versions.

Note that Hazelcast 5.4.0 requires Java 17.

Fixes #199

### CODE REVIEW

File: .github/workflows/ci-5.x.yml [hunk 1]
```diff
@@ -12,26 +12,36 @@ jobs:
   CI:
     strategy:
       matrix:
+        os: [ ubuntu-latest ]
+        jdk: [ 17 ]
```
Comment (vietj) [inline .github/workflows/ci-5.x.yml:16]: 21

File: .github/workflows/ci-5.x.yml [hunk 2]
```diff
@@ -12,26 +12,36 @@ jobs:
...
+      hz: ${{ matrix.hz }}
     secrets: inherit
   IT:
     strategy:
       matrix:
+        os: [ ubuntu-latest ]
+        jdk: [ 17 ]
+        hz: [ 5.4.0, 5.5.0 ]
         include:
           - os: ubuntu-latest
```
Comment (vietj) [inline .github/workflows/ci-5.x.yml:36]: could be removed

File: .github/workflows/ci-5.x.yml (hunk 1 above)
Comment (frant-hartm) [reply .github/workflows/ci-5.x.yml]: Done in 490ba45

**Review by tsegismont — COMMENTED** [review]
LGTM, thank you @frant-hartm 

The failures are due to a build issue on the master branch, so I'll wait for it to be solved. Then we can start the jobs again and see if everything goes well with these changes.

**Pull request merged**